presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -283,9 +286,10 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateTask(
           queryCtx->get<int32_t>(kMaxDriversPerTask.data(), maxDriversPerTask_);
       concurrentLifespans = queryCtx->get<int32_t>(
           kConcurrentLifespansPerTask.data(), concurrentLifespansPerTask_);
-      // Zero concurrent lifespans means 'unlimited'.
+      // Zero concurrent lifespans means 'unlimited', but we still limit the
+      // number to some reasonable one.
       if (concurrentLifespans == 0) {
-        concurrentLifespans = std::numeric_limits<uint32_t>::max();
+        concurrentLifespans = kMaxConcurrentLifespans;
```

Comment:
Do we run into any problems because of this? We hard code this limit here?

Reply:
We hardcode in in the constant in the beginning of the file.
We pre-allocate the slots for drivers running grouped execution in this way:
```
self->drivers_.resize(
        self->numDriversPerSplitGroup_ * self->concurrentSplitGroups_);

```
concurrentSplitGroups_ being zero means 'unlimited', but we cannot pre-allocate so many slots in the vector.
Nor it makes much sense to run so many drivers as the same time.
So this change limits it to something reasonable, say 16.